packages/astro/src/@types/astro.ts

```diff
@@ -2408,6 +2408,137 @@ export interface AstroUserConfig {
+		 * @name experimental.tsconfig
+		 * @type {object}
+		 * @default `undefined`
+		 * @version 4.15.0
+		 * @description
+		 *
+		 * Enables a streamlined way of configuring TypeScript in your project for better DX when setting up Astro for Intellisense, type-checking, and more.
+		 *
+		 * This option removes the need for a separate `src/env.d.ts` file to provide Vite's client types to your project. Instead, `/.astro/tsconfig.json` will be autogenerated with linked types for you. Notably, this fixes an existing issue with `astro check` false positives by excluding `outDir` by default.
+		 *
+		 * To enable this option in an existing project, you will need to update your existing `tsconfig.json` to reference the new file, and also potentially your `astro.config.mjs` as some items have now moved to Astro configuration options.
+		 *
+		 * Your `tsconfig.json` file must now extend at least the new `/.astro/` file:
+		 *
+		 * ```json title="tsconfig.json" ins="'./.astro/tsconfig.json'"
+		 * {
+		 * 	 extends: ['astro/tsconfigs/base', './.astro/tsconfig.json']
+		 * }
+		 * ```
+		 *
+		 * If you have `include` or `exclude` inside your root `tsconfig.json`, move them to your Astro config:
+		 *
+		 * ```json title="tsconfig.json" del={2-3}
+		 * {
+		 *   include: ['foo'],
+		 *   exclude: ['bar']
+		 * }
+		 * ```
+		 *
+		 * ```js title="astro.config.*" ins={6-7}
+		 * import { defineConfig } from 'astro/config'
+		 *
+		 * export default defineConfig({
+		 *   experimental: {
+		 *    tsconfig: {
+		 *      include: ['foo'],
+		 *      exclude: ['bar']
+		 *    }
+		 *   }
+		 * })
+		 * ```
+		 */
+		tsconfig?: {
+			/**
+			 * @docs
+			 * @name experimental.tsconfig.include
+			 * kind: h4
+			 * @type {Array<string>}
+			 * @default `undefined`
+			 * @version 4.15.0
+			 * @description
+			 *
+			 * If you have `include` your root `tsconfig.json`, you'll need to move it to your Astro config:
+			 *
+			 * ```json title="tsconfig.json" del={2}
+			 * {
+			 *   include: ['foo']
+			 * }
+			 * ```
+			 *
+			 * ```js title="astro.config.*" ins={6}
+			 * import { defineConfig } from 'astro/config'
+			 *
+			 * export default defineConfig({
+			 *   experimental: {
+			 *    tsconfig: {
+			 *      include: ['foo']
+			 *    }
+			 *   }
+			 * })
+			 * ```
+			 */
+			include?: Array<string>;
+
+			/**
+			 * @docs
+			 * @name experimental.tsconfig.exclude
+			 * @type {Array<string>}
+			 * @default `undefined`
+			 * @version 4.15.0
+			 * @description
+			 *
+			 * If you have `exclude` your root `tsconfig.json`, you'll need to move it to your Astro config:
+			 *
+			 * ```json title="tsconfig.json" del={2}
+			 * {
+			 *   exclude: ['bar']
+			 * }
+			 * ```
+			 *
+			 * ```js title="astro.config.*" ins={6}
+			 * import { defineConfig } from 'astro/config'
+			 *
+			 * export default defineConfig({
+			 *   experimental: {
+			 *    tsconfig: {
+			 *      exclude: ['bar']
+			 *    }
+			 *   }
+			 * })
+			 * ```
```

Comment:
``````suggestion
			 *
			 * Specifies an array of filenames or patterns that should be skipped when resolving [`include`](#include). It only excludes items that would otherwise be checked by your `include` settings, and cannot be used to prevent other files from being checked. See [the official TypeScript documentation](https://www.typescriptlang.org/tsconfig/#exclude) for more details.
			 *
			 * ```js title="astro.config.*" {6}
			 * import { defineConfig } from 'astro/config'
			 *
			 * export default defineConfig({
			 *   experimental: {
			 *    tsconfig: {
			 *      include: ['src/**/*'],
			 *      exclude:['src/exclude/**/*']
			 *    }
			 *   }
			 * })
			 * ```
``````
From the definition at the link it looks like whatever is excluded must be something that was going to otherwise be included. Can you update the example to make sense, given that?